### PR TITLE
Updated docs with new IntelliJ IDE story

### PR DESCRIPTION
**Pull request by mit-mit**
For details, see: https://github.com/flutter/flutter-intellij/issues/233

### CODE REVIEW

File: intellij-ide.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+To create a new Flutter IntelliJ project with a fresh set of Flutter template
+files:
+
+1. In IntelliJ, click **Create New Project** from the 'Welcome' window or
+**File>New>Project...** from the main IDE window
+1. Select **Flutter** in the menu, and click **Next**
+1. Enter your desired **Project name** and **Project location**
+1. Click **Finish**
+
+### Creating a new projects from existing source code
```
Comment (sethladd) [inline intellij-ide.md]: "From an existing codebase"

File: intellij-ide.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+files:
+
+1. In IntelliJ, click **Create New Project** from the 'Welcome' window or
+**File>New>Project...** from the main IDE window
+  - **Note**: Do *not* use the **New>Project from existing sources...** option for Flutter projects
+1. Select **Flutter** in the menu, and click **Next**
+1. Under **Project location** enter, or browse to, the directory holding your existing Flutter source code files
+1. Click **Finish**
+
+## Editing code, and viewing code problems
```
Comment (sethladd) [inline intellij-ide.md]: Maybe split these up. They seem like two different headers/sections. I'd imagine we want to show a screenshot of the "problems view" and explain the difference between an error and a warning

File: intellij-ide.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+The Dart plugin has performs code analysis that enables:
```
Comment (sethladd) [inline intellij-ide.md]: s/has/can

File: intellij-ide.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+* Viewing all current source code problems (**View>Tool Windows>Dart Analysis**)
+
+## Running and Debugging
+
+### Selecting a target
+
+When a Flutter project is open in IntelliJ, you should see a set of Flutter
+specific buttons on the right-hand side of the toolbar.
+
+* Locate the **Flutter Device Selection** drop-down button. This will show a
```
Comment (sethladd) [inline intellij-ide.md]: screenshot when we're talking about a UI element in the IDE?

File: intellij-ide.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+ list of available targets. The current selection determines which target your
+ app will be started on.
+
+* When you connect devices, or start simulators, additional entries will appear.
+
+### Run app in release mode
+
+* Click the **Play icon** in the toolbar, or invoke **Run>Run**
```
Comment (sethladd) [inline intellij-ide.md]: screenshots FTW

File: intellij-ide.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+* The bottom **Run** pane will show logs output
+
+### Run app in debug mode
+
+* If desired, set breakpoints in your source code
```
Comment (sethladd) [inline intellij-ide.md]: Use periods at the end of these sentences.

File: intellij-ide.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+To open Observatory:
+
+* Run your app in debug mode
+
+* In the Debug panel at the bottom, click the **Arrow icon** ('Open Observatory')
+
+## Fast edit and refresh development cycle
+
+Flutter offers a best-in-class developer cycle enabling you to see the effect of
+your changes near instantly!
```
Comment (sethladd) [inline intellij-ide.md]: Google style would say drop the exclamation point

File: intellij-ide.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+* Run your app in debug mode
+
+* Make desired changes in your source code (without stopping the debugger!)
```
Comment (sethladd) [inline intellij-ide.md]: drop the exclamation mark

File: faq.md [hunk 9]
```diff
@@ -287,16 +287,8 @@ possible to create your own set of widgets using our base classes.
 
 ## Does Flutter work with any editors or IDEs?
```
Comment (sethladd) [inline faq.md]: We probably want to add a FAQ for "what happened to Atom?"

File: intellij-ide.md [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+ usages (**Edit>Find>Find Usages**)
+
+* Viewing all current source code problems (**View>Tool Windows>Dart Analysis**)
+
+## Running and Debugging
+
+### Selecting a target
+
+When a Flutter project is open in IntelliJ, you should see a set of Flutter
+specific buttons on the right-hand side of the toolbar.
```
Comment (pq) [inline intellij-ide.md]: Meta-point: should we have screenshots?

File: intellij-ide.md [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+
+### Run app in release mode
+
+* Click the **Play icon** in the toolbar, or invoke **Run>Run**
+* The bottom **Run** pane will show logs output
+
+### Run app in debug mode
+
+* If desired, set breakpoints in your source code
+* Click the **(De)bug) icon** in the toolbar, or invoke **Run>Debug**
```
Comment (pq) [inline intellij-ide.md]: Extra parens?

File: intellij-ide.md [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+* The bottom **Debugger** pane will show Stack Frames and Variables
+* The bottom **Console** pane will show detailed logs output
```
Comment (pq) [inline intellij-ide.md]: Long run I think screenshots would really help...

File: intellij-ide.md (hunk 3 above)
Comment (qchong) [inline intellij-ide.md]: You can just say "The Dart plugin performs...".

File: debugging.md [hunk 13]
```diff
@@ -13,12 +13,10 @@ applications.
...
 
-Before running your applications, test your code with `flutter
-analyze`. This tool (which is a wrapper around the `dartanalyzer`
-tool) will analyze your code and help you find possible mistakes.
-If you're using the
-[Flutter plugin for Atom](https://atom.io/packages/flutter), this
-is already happening for you.
+Before running your applications, test your code with `flutter analyze`. This
+tool (which is a wrapper around the `dartanalyzer` tool) will analyze your code
+and help you find possible mistakes. If you're using the [Flutter plugin for
```
Comment (mit-mit) [inline debugging.md]: Yup, works fine. But it's harder (for a human) to read, so reformatted.

File: faq.md [hunk 14]
```diff
@@ -25,7 +25,7 @@ for text)
 framework)
 * a set of Material Design widgets (optional, you can bring-your-own
 widgets)
-* command-line tools and a plugin for Atom
+* command-line tools and a plugins for [IntelliJ IDEA](https://www.jetbrains.com/idea/)
```
Comment (mit-mit) [inline faq.md]: done

File: faq.md (hunk 9 above)
Comment (mit-mit) [inline faq.md]: I don't think that makes sense on the medium to long term horizon (hopefully very fast the amount of users who even knew of that existing will be tiny fraction).

I suggest a email announcement to the dev group is a better way of handling it.

File: faq.md [hunk 15]
```diff
@@ -287,16 +287,8 @@ possible to create your own set of widgets using our base classes.
...
-[Flutter plugin for Atom](https://atom.io/packages/dartlang). Today, it
-can syntax highlight, code
-complete, refactor, launch apps, create new apps from a template, show type
-hierarchies, jump to definition, and more.
-
-There is also a supported
-[Dart plugin for JetBrains editors](https://plugins.jetbrains.com/plugin/6351)
-like IntelliJ and WebStorm, though this plugin does not have any specific
-Flutter features.
+We are building plugins for [IntelliJ IDEA](https://www.jetbrains.com/idea/),
```
Comment (mit-mit) [inline faq.md]: done

File: intellij-ide.md [hunk 16]
```diff
@@ -0,0 +1,162 @@
+---
+layout: page
+title: Developing apps in the IntelliJ IDE
+sidebar: home_sidebar
+permalink: /intellij-ide/
+---
+
+Our custom [IntelliJ IDEA](https://www.jetbrains.com/idea/) plugins provides a
```
Comment (mit-mit) [inline intellij-ide.md]: done

File: intellij-ide.md [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+{:toc}
+
+## Installation and setup
+
+Please follow the [Flutter setup](/setup/) instructions to install IntelliJ and
+the Dart and Flutter plugins.
+
+### Updating the plugins<a name="updating"/>
+
+Updates to the plugins will be shipped on a regular basis. You should get
```
Comment (mit-mit) [inline intellij-ide.md]: No; usually plugins are updated infrequently, every 4 months as a new roll of IJ happens. We want to set the expectation that our plugin will be updated much, much more frequently than that.

Once we get a bit more mature we will slow down, and can remove this.

File: intellij-ide.md [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+prompted in IntelliJ when an update is available.
+
+To check for updates manually:
+
+1. Open preferences (**IntelliJ IDEA>Check for Updates...** on macOS, **Help>Check for Updates...** on Linux)
+1. If `dart` or `flutter-intellij` are listed, update them
+
+## Creating projects
+
+### Creating a new project from templates
```
Comment (mit-mit) [inline intellij-ide.md]: done

File: intellij-ide.md [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+
+To create a new Flutter IntelliJ project with a fresh set of Flutter template
```
Comment (mit-mit) [inline intellij-ide.md]: done

File: intellij-ide.md (hunk 1 above)
Comment (mit-mit) [inline intellij-ide.md]: left unchanged, codebase seems like an oddly specific word compared to the much more generic source code

File: intellij-ide.md (hunk 2 above)
Comment (mit-mit) [inline intellij-ide.md]: No, I think they are one section as really the key thing we want to train users to understand is that the problems view is a live view, and not some set of errors from a compilation step

File: intellij-ide.md (hunk 3 above)
Comment (mit-mit) [inline intellij-ide.md]: done

File: intellij-ide.md (hunk 2 above)
Comment (mit-mit) [inline intellij-ide.md]: Screenshots: yes, need that here and in a few other places. Not sure I will get to them for this first changeset though

File: intellij-ide.md (hunk 4 above)
Comment (mit-mit) [inline intellij-ide.md]: added todo for later

File: intellij-ide.md (hunk 5 above)
Comment (mit-mit) [inline intellij-ide.md]: added todo for later

File: intellij-ide.md (hunk 6 above)
Comment (mit-mit) [inline intellij-ide.md]: done (thoughout)

File: intellij-ide.md (hunk 7 above)
Comment (mit-mit) [inline intellij-ide.md]: done

File: intellij-ide.md (hunk 8 above)
Comment (mit-mit) [inline intellij-ide.md]: done

File: getting-started.md [hunk 20]
```diff
@@ -49,14 +49,16 @@ contains the `pubspec.yaml` file for your project).
...
+Alternatively, if you are using the [IntelliJ
+IDEA](https://www.jetbrains.com/idea/) IDE with the [Flutter
+plugins](/setup/#plugins), you can start your Flutter app from there:
+
+1. In IntelliJ, click **Create New Project** from the 'Welcome' window or
+**File>New>Project...** from the main IDE window
+1. Select **Flutter** in the menu, and click **Next**
+1. Under **Project location** enter, or browse to, the directory holding your new Flutter app
+1. Click **Finish**
+1. Click the **(De)bug icon** in the toolbar, or invoke the menu item Run>Debug
```
Comment (mit-mit) [inline getting-started.md]: The icon is a 'bug' icon, so I was trying to explain that, and guess that did not work :-). Removed, and will add screenshots (later).

File: intellij-ide.md [hunk 21]
```diff
@@ -0,0 +1,162 @@
...
+layout: page
+title: Developing apps in the IntelliJ IDE
+sidebar: home_sidebar
+permalink: /intellij-ide/
+---
+
+Our custom [IntelliJ IDEA](https://www.jetbrains.com/idea/) plugins provides a
+fully integrated development experience in the IntelliJ IDE.
+
+* TOC Placeholder
```
Comment (mit-mit) [inline intellij-ide.md]: jekyll magic that generated an actual TOC

File: intellij-ide.md (hunk 10 above)
Comment (mit-mit) [inline intellij-ide.md]: yes, added TODOs for these

File: intellij-ide.md (hunk 12 above)
Comment (mit-mit) [inline intellij-ide.md]: yes, todo added

File: intellij-ide.md (hunk 17 above)
Comment (sethladd) [inline intellij-ide.md]: What does "regular basis" mean?

I'm reacting to a promise/expectation that we're setting here, which I don't think we need to or should make. Under promise and over deliver.

**Comment by mit-mit** [comment]
Everything is relative, so more often than the standard which is every four
months.

**Comment by sethladd** [comment]
lgtm

**Pull request merged**